### PR TITLE
Redefining default settings.

### LINKED ISSUE #94: recap.conf enables FULLSTATUS by default causing email spam (bad config)
The recap.conf configuration has this enabled/on by default:
https://github.com/rackerlabs/recap/blob/master/src/recap.conf#L54
```
# Use "service httpd fullstatus"
# Please ensure that the Apache is configured to allow access to the
# http://localhost/server-status URL before enabling this option
USEFULLSTATUS=yes
```

The internal default for this variable is no:
https://github.com/rackerlabs/recap/blob/master/src/recap#L66
```
USEFULLSTATUS="no"
```

This is broken in a few ways:

1. The code uses 'apachectl fullstatus' not 'service httpd fullstatus' as documented
1. The default external config does not mirror the internal defaults; all out of the box configs should mirror the application's default behaviour, not alter it
1. By enabling this, the system keeps running the command when Apache is not installed and/or running and emailing root@localhost by default every single time it runs

The config file should mirror the internal default and be set to =no to work correctly out of the box on a server that does not have Apache installed or does not have it running. As is, it's spamming my root@localhost with the error output every 10 minutes (default cron) when it runs. Thanks!

### PR DESCRIPTION
**Pull request by tonyskapunk**
### tl;dr

- EMAIL_ON_ERROR, EMAIL_ON_ERROR_SUBJECT, EMAIL_ON_ERROR_RECIPIENTS - removed(not used)
- MAXLOAD - 10*cpu(dynamic by default, was 1000)
- LOG_EXPIRY - default 15(by default was 0, but set to 15 in the recap conf)
- ROTATE, ROTATELOGS - removed(not used)
- USEDF - default is now enabled(by default was disabled, but enabled in the recap conf)
- USEFULLSTATUS - Functionality reimplemented to avoid being tied to a web browser.
- USEMYSQL - disabled(by default was disabled, but enabled in the recap conf)
- USEMYSQLPROCESSLIST - disabled(by default was disabled, but enabled in the recap conf)
- USENETSTAT - default is enabled(by default was disabled, but enabled in the recap conf)
- USESAR - default is enabled(by default was disabled)
- OPTS_PSTREE - default is "-p"(by default was empty '')
- OPTS_STATUSURL - default is "http://localhost:80/"(this is a new option used by USEFULLSTATUS)

- Snapshots file name now matches regular logs and backups.

---
**Detailed notes of changes**
---

### Options used by the recap scripts
- BACKUP_ITEMS - Used by recap and recaplog(default: "fdisk mysql netstat ps pstree resources"). fixed in recap.conf, added to man. See NOTE 1.
- BASEDIR - used in recap and recaplog(default: "/var/log/recap"). was not defined as a setting that can be overriden, has been added to recap.c\
onf, added to man.
- EMAIL_ON_ERROR, EMAIL_ON_ERROR_SUBJECT, EMAIL_ON_ERROR_RECIPIENTS - not used, have been removed.
- LOG_COMPRESS - used in recaplog(default: 1), fixed in recap.conf, added to recaplog.8
- LOG_EXPIRY - used in recaplog(OLD default:0, NEW default: 15), fixed in recap.conf, added to recaplog.8
- MAILTO - used in recap(default: ""). fixed in recap.conf.
- MAXLOAD - used in recap(old default: "1000", new default is dynamic 10 times the number of the CPUs identified by the kernel). fixed in recap.\
conf, added to man.
- MIN_FREE_SPACE - used in recap(default: 0). fixed in recap.conf, added to man. (Minimal required space in MB)
- ROTATE - This setting is not in use any longer, has been removed from all the places where it was referenced.
- ROTATELOGS - Same as above this was not in use and has been removed.
### Reports
- USEDF - used in recap(OLD default: no, NEW default: yes). fixed in recap.conf, added to man.
- USEFDISK - used in recap(default: no). fixed in recap.conf, added to man.
- USEFULLSTATUS - used in recap(default: no). fixed in recap.conf, updated man page. NOTE: Reimplemented the functionality to avoid apache httpd\
 specific, now depends on curl instead of httpd/apache.
- USEINNODB - used in recap(default: no). fixed in recap.conf, added to man.
- USEMYSQL - used in recap(default: no). fixed in recap.conf, man updated. NOTE: This was enabled in the config previously.
- USEMYSQLPROCESSLIST - used in recap(default: no). fixed in recap.conf, man updated. NOTE: This was enabled in the config previously.
- USENETSTAT - used in recap(OLD default: no, NEW default: yes). fixed in recap.conf, changed in recap, man updated.
- USENETSTATSUM - used in recap(default: no). fixed in recap.conf, man updated.
- USEPS - used in recap(default: yes). fixed in recap.conf.
- USEPSTREE - used in recap(default: no). fixed in recap.conf, man updated.
- USERESOURCES - used in recap(default: yes). includes at leasts: uptime, free, vmstat, iostat, iotop, additional resources are included if enab\
led. fixed in recap.conf
- USESAR - used in recap(OLD default: no, NEW default: yes). fixed in recap.conf. People has requested this to be turned on("yes") by default, m\
an has been updated to reflect this. sysstat is already a depencency in the .spec file.
- USESARQ - used in recap(default: no). takes care of queue lenght(load) data (sar -q). recap.conf sets it to no as well, needs to be commented \
out in recap.conf.
- USESARR - used in recap(default: no). takes care of memory data (sar -r). recap.conf sets it to no as well, needs to be removed from recap.con\
f.
- USESLAB - used in recap(default: no). fixed in recap.conf, added to man.

### Command line Options used by the tools generating the reports

- DOTMYDOTCNF - used in recap(default: /root/.my.cnf). fixed in recap.conf, added to man.
- MYSQL_PROCESS_LIST - used in recap(default: table). fixed in recap.conf, added to man.
- OPTS_DF - used in recap(default: "-x nfs"). fixed in recap.conf.
- OPTS_FDISK - used in recap(default: "-l"). fixed in recap.conf, see NOTE 2.
- OPTS_FREE - used in recap(default: ""). fixed in recap.conf.
- OPTS_IOSTAT - used in recap(default: "-t -x 1 3"). fixed in recap.conf.
- OPTS_IOTOP - used ony by recap(default: "-b -o -t -n 3") fixed in recap.
- OPTS_NETSTAT - used in recap(default: "-ntulpae"). fixed in recap.conf.
- OPTS_NETSTAT_SUM - used in recap(default: "-s"). fixed in recap.conf.
- OPTS_PS - only used by recap(default: "auxfww"). fixed in recap.conf
- OPTS_PSTREE - used in recap(OLD default: "", NEW default: "-p"). Adding "-p" to include PID, fixed in recap.conf. updated in man.
- OPTS_STATUSURL - new variable used to avoid the use of apachectl, this is the URL that needs to be checked for the full status, default is "ht\
tp://localhost:80/", added to man.
- OPTS_VMSTAT - used in recap(default: "-S M 1 3"). fixed in recap.conf, updated in man.

### Notes

1. BACKUP_ITEMS either will have to be created dynamically from the options enabled, e.g. USE___ or the USE__ will be enabled from the BACKUP_IT\
EMS.  IF one or the other is missing/disabled there will be inconsistencies. This is a TODO item(needs an issue created for this), that will be \
addressed in another PR.

2. FDISK needs love, needs to be more dynamic and use parted/fdisk, but this limits the options, it will have to be only "-l". This isa TODO ite\
m(needs an issue created for this), this is a feature.

#### recap
- LOG_SUFFIX=$(date +%Y%m%d-%H%M%S)
- PATH=/bin:/usr/bin:/sbin:/usr/sbin
- DATE=`date +%Y-%m-%d_%H:%M:%S` ## In the snapshot functionality, this provides inconsistencies with LOG_SUFFIX, removing and using LOG_SUFFIX \
instead.
- SNAPSHOT="no"
- BACKUP="no"
- LOCKFILE="/var/lock/recap.lock"
- BACKUP_ITEMS="fdisk mysql netstat ps pstree resources"
- MAXLOAD="1000"
- OUTPUTDIR_MODE="750"

#### recaplog
- PATH=/bin:/usr/bin:/sbin:/usr/sbin
- LOCKFILE="/var/lock/recaplog.lock"
- LOG_COMPRESS=1
- LOG_EXPIRY=0
- BACKUP_ITEMS="fdisk mysql netstat ps pstree resources"
- LOGFILE="${BASEDIR}/recaplog.log"
- ## These needs to be removed as recaplog does not send email anymore
- MAIL_ON_ERROR=1
- EMAIL_ON_ERROR_SUBJECT="recaplog error"
- EMAIL_ON_ERROR_RECIPIENTS="root@localhost"

### Setting Dependencies:

- USERESOURCES
  - USEDF
  - USESAR
  - USESARQ
  - USESARR
  - USEFULLSTATUS

- USENETSTAT
  - USENETSTATSUM

- USEMYSQL
  - USEMYSQLPROCESSLIST
  - USEINNODB
  - DOTMYDOTCNF
  - MYSQL_PROCESS_LIST

### Other changes
1. STATUSURL - This was introduced to be more generic with USEFULLSTATUS, this will allow to query any webserver that supports status(httpd, ngi\
nx)
1. Snapshots file name matches regular logs and backups, example of the differences:

```bash
## Regular format on recap
# ls -ltr /var/log/recap/ | tail -4
-rw-r--r--. 1 root root    820 Jun  9 14:30 pstree_20170609-143001.log
-rw-r--r--. 1 root root   6314 Jun  9 14:30 netstat_20170609-143001.log
-rw-r--r--. 1 root root    297 Jun  9 14:30 fdisk_20170609-143001.log
-rw-r--r--. 1 root root   5127 Jun  9 14:30 mysql_20170609-143001.log

## Backups use the same log formatting as recap regular logs:
# ls -ltr /var/log/recap/backups | tail -4
-rw-r--r--. 1 root root 6191 Jun  9 14:27 netstat_20170609-142001.log
-rw-r--r--. 1 root root 8070 Jun  9 14:27 ps_20170609-142001.log
-rw-r--r--. 1 root root  785 Jun  9 14:27 pstree_20170609-142001.log
-rw-r--r--. 1 root root 8407 Jun  9 14:27 resources_20170609-142001.log

## Snapshots uses a completely different formatting:
# ls -ltr /var/log/recap/snapshots/ | tail -4
-rw-r--r--. 1 root root  812 Jun  9 14:27 pstree.log_snapshot_2017-06-09_14:27:43
-rw-r--r--. 1 root root 6314 Jun  9 14:27 netstat.log_snapshot_2017-06-09_14:27:43
-rw-r--r--. 1 root root  297 Jun  9 14:27 fdisk.log_snapshot_2017-06-09_14:27:43
-rw-r--r--. 1 root root 5127 Jun  9 14:27 mysql.log_snapshot_2017-06-09_14:27:43
```

Now uses:
```bash
# ls -ltr /var/log/recap/snapshots/ | tail -4
-rw-r--r--. 1 root root  899 Jun  9 14:40 pstree_20170609-144025.log_snapshot
-rw-r--r--. 1 root root 6437 Jun  9 14:40 netstat_20170609-144025.log_snapshot
-rw-r--r--. 1 root root  297 Jun  9 14:40 fdisk_20170609-144025.log_snapshot
-rw-r--r--. 1 root root 5127 Jun  9 14:40 mysql_20170609-144025.log_snapshot
```

1. recaptool now makes use of BASEDIR to align with recap and recaplog.

---

Fix #94

### CODE REVIEW

File: src/recap [hunk 1]
```diff
@@ -233,28 +233,28 @@ print_sar() {
...
 
-# print the output of "apachectl fullstatus" to the specified file
-print_httpd_fullstatus() {
+# Print the output of the web server status page to the specified file
+print_http_fullstatus() {
   local LOGFILE="$1"
-  echo "Apache Status report" >> "${LOGFILE}"
-  apachectl fullstatus >> "${LOGFILE}"
+  echo "Web Status report" >> "${LOGFILE}"
+  curl -Ls "${OPTS_STATUSURL}" 2>/dev/null >> "${LOGFILE}"
```
Comment (troyengel) [inline src/recap]: Should we make a new 'OPTS_CURL="-Ls"' here? I thought "what if the user has https://... and needs to use the -k (ignore cert mismatch) with curl?" Technically you could add this to OPTS_STATUSURL (="-k https://...") but that is sort of ugly.

Comment (tonyskapunk) [inline src/recap]: Indeed. There are more use cases where different options can be useful, a quick one is `-v` so I've added that.

**Comment by tonyskapunk** [comment]
Had to rebase due to a commit in `development` but kept the newer commit separated from the initial one, 8f6317a includes the newly added `OPTS_CURL` and the consistency with variable definition.

**Comment by tonyskapunk** [comment]
Ok, it has been 2w since this PR was opened, the last commit covers the comments by @troyengel , I'm merging this into development branch now.
